packages/mgt-components/src/components/mgt-search-results/mgt-search-results.ts

```diff
@@ -0,0 +1,893 @@
+ */
+@customElement('search-results')
+export class MgtSearchResults extends MgtTemplatedComponent {
+  /**
+   * Array of styles to apply to the element. The styles should be defined
+   * user the `css` tag function.
+   */
+  static get styles() {
+    return styles;
+  }
+
+  protected get strings() {
+    return strings;
+  }
+
+  private _queryString: string;
+  /**
+   * The query to send to Microsoft Search
+   *
+   * @type {string}
+   * @memberof MgtSearchResults
+   */
+  @property({
+    attribute: 'query-string',
+    reflect: true,
+    type: String
+  })
+  public get queryString(): string {
+    return this._queryString;
+  }
+  public set queryString(value: string) {
+    if (this._queryString !== value) {
+      this._queryString = value;
+      this._currentPage = 1;
+      this.setLoadingState(true);
+      this.requestStateUpdate(true);
+    }
+  }
+
+  /**
+   * Query template to use in complex search scenarios
+   * Query Templates are currently supported only on the beta endpoint
+   */
+  @property({
+    attribute: 'query-template',
+    type: String
+  })
+  public queryTemplate: string;
+
+  /**
+   * One or more types of resources expected in the response.
+   * Possible values are: list, site, listItem, message, event,
+   * drive, driveItem, externalItem.
+   *
+   * @type {string}
+   * @memberof MgtSearchResults
+   */
+  @property({
+    attribute: 'entity-types',
+    converter: value => {
+      return value.split(',').map(v => v.trim());
+    },
+    type: String
+  })
+  public entityTypes: string[];
+
+  /**
+   * The scopes to request
+   *
+   * @type {string[]}
+   * @memberof MgtSearchResults
+   */
+  @property({
+    attribute: 'scopes',
+    converter: (value, type) => {
+      return value ? value.toLowerCase().split(',') : null;
+    },
+    reflect: true
+  })
+  public scopes: string[] = [];
+
+  /**
+   * Content sources to use with External Items
+   *
+   * @type {string[]}
+   * @memberof MgtSearchResults
+   */
+  @property({
+    attribute: 'content-sources',
+    converter: (value, type) => {
+      return value ? value.toLowerCase().split(',') : null;
+    },
+    reflect: true
+  })
+  public contentSources: string[] = [];
+
+  /**
+   * Api version to use for request
+   *
+   * @type {string}
+   * @memberof MgtSearchResults
+   */
+  @property({
+    attribute: 'version',
+    reflect: true,
+    type: String
+  })
+  public version: string = 'v1.0';
+
+  /**
+   * Specifies the offset for the search results.
+   * Offset 0 returns the very first result.
+   *
+   * @type {number}
+   * @memberof MgtSearchResults
+   */
+  public get from(): number {
+    return (this.currentPage - 1) * this.size;
+  }
+
+  /**
+   * The size of the page to be retrieved.
+   * The maximum value is 1000.
+   *
+   * @type {number}
+   * @memberof MgtSearchResults
+   */
+  @property({
+    attribute: 'size',
+    reflect: true,
+    type: Number
+  })
+  public size: number = 10;
```

Comment:
@sebastienlevert since the max = 1000, when you supply a value >1000 can we have the setter just default it to 1000?

Reply:
Good idea!